lib/smack/NormalizeLoops.cpp

```diff
@@ -0,0 +1,134 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+
+//
+// This pass normalizes loop structures to allow easier generation of Boogie
+// code when a loop edge comes from a branch. Specifically, a forwarding block
+// is added between a branching block and the loop header.
+//
+
+#include "smack/NormalizeLoops.h"
+#include "smack/Naming.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/ValueSymbolTable.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/Analysis/LoopInfo.h"
+#include "llvm/IR/Dominators.h"
+#include "llvm/Support/raw_ostream.h"
+#include <map>
+#include <set>
+
+namespace smack {
+
+using namespace llvm;
+
+// Register LoopInfo
+void NormalizeLoops::getAnalysisUsage(AnalysisUsage& AU) const {
+  AU.addRequired<LoopInfoWrapperPass>();
+}
+
+/**
+ * Creates a basic block in \param func which unconditionally branches to
+ * \param target.
+ */
+BasicBlock* makeForwardingBlock(BasicBlock* target, Function* func) {
+  auto& context = target->getContext();
+  BasicBlock* result = BasicBlock::Create(context, "forwarder", func);
+  BranchInst* branch = BranchInst::Create(target);
+  result->getInstList().push_back(branch);
+  return result;
+}
+
+/**
+ * Splits any edge between \param BB and \param headerBlock in a loop body,
+ * by adding a forwarding block between \param BB and \param headerBlock. The
+ * \param blockMap is updated to map between \param BB and the newly created
+ * forwarding block.
+ */
+void splitBlock(BasicBlock* BB, BasicBlock* headerBlock, Function* func, std::map<BasicBlock*,BasicBlock*>& blockMap) {
```

Comment:
Please split this argument list into two lines.

Reply:
Done.